routing/pathfind.go

```diff
@@ -536,6 +537,23 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		}
 	}
 
+	// Build a preliminary destination hop structure to obtain the payload
+	// size.
+	var mpp *record.MPP
+	if r.PaymentAddr != nil {
+		mpp = record.NewMPP(amt, *r.PaymentAddr)
+	}
+
+	finalHop := &route.Hop{
```

Comment:
i see what you mean now about using the hops directly, very cool!

Reply:
Yes, so initially i planned to store these in the map. Maybe we'll do it in the future.